os/net/app-layer/mqtt/mqtt.h

```diff
@@ -113,9 +131,23 @@
 
 #define MQTT_FHDR_SIZE 1
 #define MQTT_MAX_REMAINING_LENGTH_BYTES 4
-#define MQTT_PROTOCOL_VERSION 3
+#if MQTT_31
 #define MQTT_PROTOCOL_NAME "MQIsdp"
+#else
+#define MQTT_PROTOCOL_NAME "MQTT"
+#endif
+
 #define MQTT_TOPIC_MAX_LENGTH 128
+
+#if MQTT_PROTOCOL_VERSION >= MQTT_PROTOCOL_VERSION_3_1_1
+#ifdef MQTT_CONF_SUPPORTS_EMPTY_CLIENT_ID
+#define MQTT_SRV_SUPPORTS_EMPTY_CLIENT_ID MQTT_CONF_SUPPORTS_EMPTY_CLIENT_ID
+#else
+#define MQTT_SRV_SUPPORTS_EMPTY_CLIENT_ID 0
+#endif
```

Comment:
Conventionally we'd add a `/* MQTT_CONF_SUPPORTS_EMPTY_CLIENT_ID */` to show which conditional is closing here